examples/ShowPlayer.cpp

```diff
@@ -85,68 +92,111 @@ int ShowPlayer::Playback(unsigned int iterations,
   return ola::EXIT_OK;
 }
 
-void ShowPlayer::SendNextFrame() {
-  DmxBuffer buffer;
-  unsigned int universe;
-  ShowLoader::State state = m_loader.NextFrame(&universe, &buffer);
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
-  }
 
-  state = RegisterNextTimeout();
+/**
+ * Start playback
+ */
+void ShowPlayer::Start() {
+  SeekTo(m_start);
+}
+
 
-  OLA_INFO << "Universe: " << universe << ": " << buffer.ToString();
-  ola::client::SendDMXArgs args;
-  m_client.GetClient()->SendDMX(universe, buffer, args);
-
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
+/**
+ * Seek to @p seek_time in the show file
+ * @param seek_time the time (in milliseconds) to seek to
+ */
+void ShowPlayer::SeekTo(const unsigned int seek_time) {
+  // Seeking to a time before the playhead's position requires moving from the
+  // beginning of the file.  This could be optimized more if this happens
+  // frequently.
+  if (seek_time < m_playback_pos) {
+    m_loader.Reset();
+    m_playback_pos = 0;
   }
+
+  // Keep reading through the show file until desired time is reached.
+  ShowEntry entry;
+  unsigned int playhead_time = m_playback_pos;
+  do {
+    ShowLoader::State state = m_loader.NextEntry(&entry);
+    switch (state) {
+      case ShowLoader::END_OF_FILE:
+        HandleEndOfFile();
```

Comment:
Shouldn't this just bomb out and not call this function during seek, otherwise if your file is 10s, your start is 20s and infinite loops you would get:
1. SeekTo(time after end of file)
1. Hit EOF
1. m_loader.reset()
1. SeekTo(time after end of file)
etc

I think it should just say it overran the file looking for the start and then:
`m_client.GetSelectServer()->Terminate();`

Reply:
That's a good point - there really isn't a valid case where it should hit EOF here.